motoman_driver/src/industrial_robot_client/joint_trajectory_action.cpp

```diff
@@ -532,18 +554,19 @@ void JointTrajectoryAction::controllerStateCB(
     return;
   }
 
-  if (!industrial_utils::isSimilar(all_joint_names_, msg->joint_names))
+  // joint torso_joint_b2 is not in msg->joint_names, so this code is commented out.
```

Comment:
Sorry for the delay on my part, but I'm in agreement with @gavanderhoorn.  Any solution should be independent of the robot type.  I can't think of a bulletproof way of knowing the robot type.  I don't fully understand why the message doesn't have `torso_joint_b2`.  Is the answer to just fix the `msg->joint_names` publisher?